src/main/java/org/apache/commons/text/similarity/JaroWinklerDistance.java

```diff
@@ -71,20 +76,7 @@ public Double apply(final CharSequence left, final CharSequence right) {
             throw new IllegalArgumentException("CharSequences must not be null");
         }
 
-        // TODO: replace the rest of the code by this in 2.0, see TEXT-104
-        //
-        // JaroWinklerSimilarity similarity = new JaroWinklerSimilarity();
-        // return 1 - similarity.apply(left, right);
-
-        final double defaultScalingFactor = 0.1;
-        final int[] mtp = matches(left, right);
-        final double m = mtp[0];
-        if (m == 0) {
-            return 0D;
-        }
-        final double j = ((m / left.length() + m / right.length() + (m - (double) mtp[1] / 2) / m)) / 3;
-        final double jw = j < 0.7D ? j : j + defaultScalingFactor * mtp[2] * (1D - j);
-        return jw;
+        return 1 - similarity.apply(left, right);
```

Comment:
I used the `JaroWinklerSimilarity` as mentioned  in the JIRA issue.

In TEXT-104, we had a regression as I had removed the `matches` method, that is now deprecated. When I reverted it, I could have also fixed the behaviour here.

Even though this is a change that could break the code of existing users, it's actually fixing a bug in the way we calculate the edit distance, making similarity and distance return different values, and the distance will be `0` for identical values, and `1` for the most distant value possible — as with other distances.

I used some Python libraries to compare the output, and the value of our `similarity` matches their similarity/metric/or distance.